tests/Utilities/MakeWithRandomValues.hpp

```diff
@@ -73,67 +89,40 @@ struct FillWithRandomValuesImpl<Variables<tmpl::list<Tags...>>,
- private:
-  std::array<T, 2> bounds_;
-
+class UniformCustomDistribution<T, Requires<std::is_integral<T>::value>>
+    : public std::uniform_int_distribution<T> {
  public:
-  using result_type = T;
-  template <typename Bound>
-  UniformCustomDistribution(Bound low, Bound high) noexcept
-      : bounds_{{static_cast<T>(low), static_cast<T>(high)}} {}
+  using std::uniform_int_distribution<T>::uniform_int_distribution;
   template <typename Bound>
-  explicit UniformCustomDistribution(std::array<Bound, 2> bounds_) noexcept
-      : bounds_{{static_cast<T>(bounds_[0]), static_cast<T>(bounds_[1])}} {}
-
-  template <class Generator>
-  T operator()(Generator& g) noexcept {
-    return ((static_cast<T>(g()) - static_cast<T>(Generator::min())) *
-                (gsl::at(bounds_, 1) - gsl::at(bounds_, 0)) /
-                static_cast<T>(Generator::max() - Generator::min()) +
-            gsl::at(bounds_, 0));
-  }
+  explicit UniformCustomDistribution(std::array<Bound, 2> arr)
+      : std::uniform_int_distribution<T>(static_cast<T>(arr[0]),
+                                         static_cast<T>(arr[1])) {}
+  using std::uniform_int_distribution<T>::operator=;
+  using std::uniform_int_distribution<T>::operator();
 };
-
-/// \ingroup TestingFrameworkGroup
-/// \brief Specialization of the generation function to independently sample the
-/// components of a std::complex<T>. Note, for convenience, the bounds are real
-/// and describe the width of the complex square to be sampled.
 template <typename T>
-struct UniformCustomDistribution<std::complex<T>> {
- private:
-  std::array<T, 2> bounds_;
-
+class UniformCustomDistribution<T, Requires<std::is_floating_point<T>::value>>
+    : public std::uniform_real_distribution<T> {
  public:
-  using result_type = std::complex<T>;
+  using std::uniform_real_distribution<T>::uniform_real_distribution;
   template <typename Bound>
-  UniformCustomDistribution(Bound low, Bound high) noexcept
-      : bounds_{{static_cast<T>(low), static_cast<T>(high)}} {}
-  template <typename Bound>
-  explicit UniformCustomDistribution(std::array<Bound, 2> bounds_) noexcept
-      : bounds_{{static_cast<T>(bounds_[0]), static_cast<T>(bounds_[1])}} {}
-
-  template <class Generator>
-  std::complex<T> operator()(Generator& g) noexcept {
-    auto val1 = ((static_cast<T>(g()) - static_cast<T>(Generator::min())) *
-                     std::real(gsl::at(bounds_, 1) - gsl::at(bounds_, 0)) /
-                     static_cast<T>(Generator::max() - Generator::min()) +
-                 std::real(gsl::at(bounds_, 0)));
-    auto val2 = ((static_cast<T>(g()) - static_cast<T>(Generator::min())) *
-                     std::real(gsl::at(bounds_, 1) - gsl::at(bounds_, 0)) /
-                     static_cast<T>(Generator::max() - Generator::min()) +
-                 std::real(gsl::at(bounds_, 0)));
-    return std::complex<T>(val1, val2);
-  }
+  explicit UniformCustomDistribution(std::array<Bound, 2> arr)
```

Comment:
I would have `const std::array<T, 2>& arr` since the type must be convertible to `T` without precision loss, otherwise this is somewhat sketchy. Taking an array of `T` forces the person constructing the object to think about conversions instead of making an implicit choice for them.

Reply:
How about with the template but without casts? Otherwise, the array typing makes this constructor more strict than the base type constructor.

Reply:
Why would that be stricter? It would be the same as far as I can tell. If we have different definitions of the same, then I think what you want is `T{arr[0]}`, which is a safe conversion

Reply:
If not templated, an `array<double,2>` will not implicitly convert to an `array<int, 2>`, but doubles are fine in the constructor for `uniform_int_distribution`. I'd like this constructor to work for all arrays of types valid for the std constructor.

Reply:
Okay, then I think you want to do `T{arr[0]}, T{arr[1]}` to get the same behavior (compiler error if there's a precision losing conversion)

Reply:
hmm.. still stricter than the underlying constructor, which has no way of preventing implicit conversion. Doing this would ensure a lot of my general tests that use double arrays for generic bounds would just call the constructor with two arguments to get the easy implicit conversion, largely just skipping the convenience array constructor. If that's the desired style, though, it's only a small change in all the user code.